### PR TITLE
completed-issue-7-user-can-flip-a-card

### PR DESCRIPTION
**Pull request by anthony-urbina**
https://user-images.githubusercontent.com/109625982/202829703-ea8f72b1-c14b-45b5-98f8-806ae9a379ae.mov

https://user-images.githubusercontent.com/109625982/202829710-d2563548-6c3a-4f52-a3e6-2a42dfaf533d.mov

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -95,34 +95,59 @@ app.get('/api/games/retrieve/:userId', (req, res, next) => {
...
+  `;
+  const params = [gameId, state];
+
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        throw new ClientError(400, 'this gameId does not exist');
+      }
+      res.status(200).json(result.rows[0].state);
+    })
```
Comment (thebearingedge) [inline server/index.js]: If you include the room id in the request body, you can broadcast this update back to the clients

```js
io.to(roomId).emit('flip-card', updatedState)
```

Then in your client code, you can listen for `'flip-card'` and update both clients at the same time.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 1 above)
Comment (anthony-urbina) [reply server/index.js]: cha chinggggg 🤑

**Comment by anthony-urbina** [comment]
https://user-images.githubusercontent.com/109625982/202864005-32f22da1-79f1-43a9-8186-abbc84d848d3.mov

File: package.json [hunk 2]
```diff
@@ -15,11 +15,15 @@
...
     "express": "^4.18.2",
+    "from": "^0.1.7",
     "jsonwebtoken": "^8.5.1",
     "jwt-decode": "^3.1.2",
+    "lodash": "^4.17.21",
+    "lodash.shuffle": "^4.2.0",
     "pg": "^8.8.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "shuffle": "^0.2.5",
```
Comment (thebearingedge) [inline package.json]: Please take a moment to `npm uninstall` whichever packages you didn't end up using.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Pull request merged**